lib/Backend.hs

```diff
@@ -26,3 +27,5 @@ class (Show a, A.FromJSON a) => Backend a where
   _readEntry :: Effects r => a -> EntryPath -> Sem r (Maybe Entry)
   _listDirectoryContents :: Effects r => a -> Path -> Sem r (Maybe DirectoryContents)
   _deleteEntry :: Effects r => a -> EntryPath -> Sem r ()
+  _validatePath :: Effects r => (HasPathSegments s segments, Each segments segments PathSegment PathSegment)
```

Comment:
I've made... smth
It works, but I'm not sure that it's intended way
I'm understanding why we need Iso & each. But do we still need `HasPathSegments` layer?
Seems I just had to add Each constraint to `HasPathSegments`. Because we always want segments contain `PathSegment`

Reply:
As discussed in the call, instead of having a `SuperPathSegmented` class, let's add an `Each` constraint to the `HasPathSegments` class:

```hs
class Each segments segments PathSegment PathSegment => HasPathSegments s segments | s -> segments where
  pathSegments :: Iso' s segments
```

Reply:
done